xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/resources/xwiki-focusedplaceholder/plugin.js

```diff
@@ -0,0 +1,268 @@
+                    h3: "xwiki-focusedplaceholder.placeholder.h3", // Heading 3
+                    h4: "xwiki-focusedplaceholder.placeholder.h4", // Heading 4
+                    h5: "xwiki-focusedplaceholder.placeholder.h5", // Heading 5
+                    h6: "xwiki-focusedplaceholder.placeholder.h6", // Heading 6
+                    main: "xwiki-focusedplaceholder.placeholder.main", // Main content
+                    nav: "xwiki-focusedplaceholder.placeholder.nav", // Navigation
+                    section: "xwiki-focusedplaceholder.placeholder.section", // Section
+                    blockquote: "xwiki-focusedplaceholder.placeholder.blockquote", // Quote
+                    dl: "xwiki-focusedplaceholder.placeholder.dl", // Definition list
+                    dt: "xwiki-focusedplaceholder.placeholder.dt", // Definition entry
+                    dd: "xwiki-focusedplaceholder.placeholder.dd", // Description
+                    figcaption: "xwiki-focusedplaceholder.placeholder.figcaption", // Caption
+                    figure: "xwiki-focusedplaceholder.placeholder.figure", // Figure
+                    ol: "xwiki-focusedplaceholder.placeholder.ol", // Ordered list
+                    ul: "xwiki-focusedplaceholder.placeholder.ul", // Unordered list
+                    menu: "xwiki-focusedplaceholder.placeholder.menu", // Menu (unordered list)
+                    li: "xwiki-focusedplaceholder.placeholder.li", // List item
+                    p: "xwiki-focusedplaceholder.placeholder.p", // Paragraph
+                    pre: "xwiki-focusedplaceholder.placeholder.pre", // Preformated
+                    caption: "xwiki-focusedplaceholder.placeholder.caption", // Title
+                    th: "xwiki-focusedplaceholder.placeholder.th", // Header
+                    td: "xwiki-focusedplaceholder.placeholder.td", // Cell
+                },
+                /* Empty inline tags (or nodeNames) that usually do not appear on screen should be ignored by default
+                 * because they might get propagated to new lines by CKEditor*/
+                ignoreIfEmpty: [
+                    "#text",
+                    "a",
+                    "abbr",
+                    "b",
+                    "bdi",
+                    "bdo",
+                    "br",
+                    "cite",
+                    "data",
+                    "dfn",
+                    "em",
+                    "i",
+                    "mark",
+                    "s",
+                    "small",
+                    "span",
+                    "strong",
+                    "time",
+                    "u",
+                    "var",
+                    "wbr",
+                    "del",
+                    "ins",
+                ],
+            };
+        },
+
+        init: function(editor) {
+
+            var focusedplaceholder = editor.plugins["xwiki-focusedplaceholder"];
+            focusedplaceholder.editor = editor;
```

Comment:
This is not right. Plugin instances are shared between editor instances. Edit ``/xwiki/bin/view/Help/Applications/Movies/Meet John Doe`` which has 2 editor instances and do this in the JavaScript console:
```
CKEDITOR.instances['Help.Applications.Movies.Code.MoviesClass_0_longText1'].plugins['xwiki-macro'].foo = 'bar'
CKEDITOR.instances['content'].plugins['xwiki-macro'].foo
```

If you need a state you should save it on the editor instance, not on the plugin instance.

Also, the preferred style is to use camelcase for variable names, so ``focusedPlaceholder``.

Reply:
I did that so that the different functions would have access to the configuration. I didn't know the plugin instances were shared, thank you!
I moved the functions definitions in the init method so the editor would be in their scope.